tests/python/pants_test/pantsd/test_pailgun_server.py

```diff
@@ -78,6 +83,46 @@ def test_process_request_thread_error(self, mock_shutdown_request):
     self.assertIs(self.mock_handler_inst.handle_error.called, True)
     mock_shutdown_request.assert_called_once_with(self.server, mock_request)
 
+  def test_process_request_thread_waits_until_timeout(self):
```

Comment:
Discussed this test offline; @blorente is going to try to rewrite this as a test which synchronises threads and checks counters, rather than relying on timing-based.

Reply:
Rewritten :) no `time.sleep`s now.